Helper/Cart.php

```diff
@@ -1357,18 +1357,28 @@ function ($item) use ($imageHelper, &$totalAmount, &$diff, $storeId, $currencyCo
                 ////////////////////////////////////
                 // Load item product object
                 ////////////////////////////////////
-                $itemProduct = $item->getProduct();
-                $customizableOptions = $this->getProductCustomizableOptions($itemProduct);
-                $itemSku = trim($item->getSku());
-
-                if ($customizableOptions) {
-                    $itemSku = $this->getProductActualSkuByCustomizableOptions($itemSku, $customizableOptions);
+                //By default this feature switch is enabled.
+                if ($this->deciderHelper->isCustomizableOptionsSupport()) {
+                    $itemProduct = $item->getProduct();
+                    $customizableOptions = $this->getProductCustomizableOptions($itemProduct);
+                    $itemSku = trim($item->getSku());
+                    
+                    if ($customizableOptions) {
+                        $itemSku = $this->getProductActualSkuByCustomizableOptions($itemSku, $customizableOptions);
+                    }
+                    
+                    $_product = $this->productRepository->get($itemSku);
```

Comment:
What if trying to catch error here and if so continue with `$itemProduct`?
What do you think about it?

Reply:
Another point, I think we shouldn't search product by sku if we didn't change it. We just added additional SQL calls without benefits.
So something like
```
                    if ($customizableOptions) {
                        $itemSku = $this->getProductActualSkuByCustomizableOptions($itemSku, $customizableOptions);
                         $_product = $this->productRepository->get($itemSku);
                    }
```

Reply:
Also, can we add productId to bugsnag record if we have any exceptions?